mmv1/third_party/terraform/services/cloudidentity/data_source_cloud_identity_group_transitive_memberships.go.erb

```diff
@@ -0,0 +1,179 @@
+group or user.
+
+For external-identity-mapped entities, the id must be a string conforming
+to the Identity Source's requirements.
+
+Must be unique within a namespace.`,
+									},
+									"namespace": {
+										Type:     schema.TypeString,
+										Computed: true,
+										Description: `The namespace in which the entity exists.
+
+If not specified, the EntityKey represents a Google-managed entity
+such as a Google user or a Google Group.
+
+If specified, the EntityKey represents an external-identity-mapped group.
+The namespace must correspond to an identity source created in Admin Console
+and must be in the form of 'identitysources/{identity_source_id}'.`,
+									},
+								},
+							},
+						},
+						"member": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `Resource name for this member.`,
+						},
+						"relation_type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The relation between the group and the transitive member. The value can be DIRECT, INDIRECT, or DIRECT_AND_INDIRECT`,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceGoogleCloudIdentityGroupTransitiveMembershipsRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+
+	result := []map[string]interface{}{}
+	membershipsCall := config.NewCloudIdentityClient(userAgent).Groups.Memberships.SearchTransitiveMemberships(d.Get("group").(string))
+	if config.UserProjectOverride {
+		billingProject := ""
+		// err may be nil - project isn't required for this resource
+		if project, err := tpgresource.GetProject(d, config); err == nil {
+			billingProject = project
+		}
+
+		// err == nil indicates that the billing_project value was found
+		if bp, err := tpgresource.GetBillingProject(d, config); err == nil {
+			billingProject = bp
+		}
+
+		if billingProject != "" {
+			membershipsCall.Header().Set("X-Goog-User-Project", billingProject)
+		}
+	}
+
+	err = membershipsCall.Pages(config.Context, func(resp *cloudidentity.SearchTransitiveMembershipsResponse) error {
+		for _, member := range resp.Memberships {
+			result = append(result, map[string]interface{}{
+				"member":               member.Member,
+				"relation_type":        member.RelationType,
+				"roles":                flattenCloudIdentityGroupTransitiveMembershipsRoles(member.Roles),
+				"preferred_member_key": flattenCloudIdentityGroupsEntityKeyList(member.PreferredMemberKey),
+			})
+		}
+
+		return nil
+	})
+	if err != nil {
+		return transport_tpg.HandleDataSourceNotFoundError(err, d, fmt.Sprintf("CloudIdentityGroupMemberships %q", d.Id()), "")
+	}
+
+	if err := d.Set("memberships", result); err != nil {
+		return fmt.Errorf("Error setting memberships: %s", err)
+	}
+	d.SetId(time.Now().UTC().String())
```

Comment:
Is this a typical way to set an ID in a datasource? Do we not want it to be something related to this datasource in particular?

Reply:
Id format changed in [c28db4e](https://github.com/GoogleCloudPlatform/magic-modules/pull/11337/commits/c28db4ed5e95535eae456a2ca2357e74d2c7eca5) to `groups/{group_id}/transitiveMemberships`